ledger/eval_test.go

```diff
@@ -1402,3 +1410,57 @@ func TestModifiedAppLocalStates(t *testing.T) {
 		assert.False(t, created)
 	}
 }
+
+// TestAppInsMinBalance checks that accounts with MaxAppsOptedIn are accepted by block evaluator
+// and do not cause any MaximumMinimumBalance problems
```

Comment:
I guess it should also be opted into 1000 assets, to make sure MaxMinBalance is correct.

Reply:
Also, the apps being opted into must have "full" schemas to get max min balance.

Reply:
> Also, the apps being opted into must have "full" schemas to get max min balance.
Agree

>I guess it should also be opted into 1000 assets, to make sure MaxMinBalance is correct.
No, because 1000 assets will consume entire MinMaxBalance alone (1000 * 0.1 = 100 that is almost 100.1 of MaxMinBalance). So a user needs to decide what he wants to store - if we want to change it, it needs to be a separate PR.

Reply:
I think that MinMaxBalance value is, essentially, a bug right now.  I believe the spec says you can opt into 1000 assets and 10 apps.  That's not true.  But I think the intent is that it is.

I guess I don't care too much if it's in this PR or another.

Reply:
The spec only says without binding it 1000 assets or 10 apps or something. I might be missing something tho
```
A transaction must be rejected if its inclusion in a block would cause an account's minimum balance to exceed the maximum minimum balance of 100100000 microalgos.
```

Reply:
It also separately says both of the asset and app limits:
"Accounts can hold up to $Assets_{\max}$ assets (1000 in this protocol)"
"The maximum number of applications that a single account may opt in to is 10."

I suppose the statement on MaxMinBalance supersedes these statements, in a sense, but I think the intent was to set MaxMinBalance so that it did not prevent the other limits but we forgot to bump it when we added apps.  I'm only guessing.